src/components/dataSet-settings.js

```diff
@@ -182,72 +198,83 @@ class DataSetSettings extends React.Component {
      */
     handleSubmitDialog = () => {
-        var specificDataSetNameKey = this.state.specificSettingName
+        var specificDataSetNameKey = this.state.specificSetting.name
         var objData = this.specificSettings
 
         const dataSetNameFilter = this.dataSetListComplete.filter(
             option => option.id === specificDataSetNameKey
         )
 
-        objData[specificDataSetNameKey] = {
-            id: specificDataSetNameKey,
-            lastUpdated: new Date().toJSON(),
-            name: dataSetNameFilter[0].name,
-            specificPeriodDSDownload: this.state.specificPeriodDSDownload,
-            specificPeriodDSDBTrimming: this.state.specificPeriodDSDBTrimming,
-        }
+        if (dataSetNameFilter.length > 0) {
+            objData[specificDataSetNameKey] = {
+                id: specificDataSetNameKey,
+                lastUpdated: new Date().toJSON(),
+                name: dataSetNameFilter[0].name,
+                periodDSDownload: this.state.specificSetting.periodDSDownload,
+                periodDSDBTrimming: this.state.specificSetting
+                    .periodDSDBTrimming,
+            }
 
-        const sumarySettings =
-            this.state.specificPeriodDSDownload === undefined
-                ? 0
-                : this.state.specificPeriodDSDownload
-        const newDataSetRow = {
-            name: dataSetNameFilter[0].name,
-            sumarySettings: sumarySettings,
-            specificPeriodDSDownload: this.state.specificPeriodDSDownload,
-            specificPeriodDSDBTrimming: this.state.specificPeriodDSDBTrimming,
-            id: specificDataSetNameKey,
-        }
+            const sumaryString = this.state.specificSetting.periodDSDownload
+            const sumarySettings =
+                this.state.specificSetting.periodDSDownload === undefined
+                    ? undefined
+                    : this.arrangeWord(sumaryString)
+            const newDataSetRow = {
+                name: dataSetNameFilter[0].name,
+                sumarySettings: sumarySettings,
+                periodDSDownload: this.state.specificSetting.periodDSDownload,
+                periodDSDBTrimming: this.state.specificSetting
+                    .periodDSDBTrimming,
+                id: specificDataSetNameKey,
+            }
 
-        this.specificSettings = objData
+            this.specificSettings = objData
 
-        const dataSetData = {
-            specificSettings: objData,
-        }
+            const dataSetData = {
+                specificSettings: objData,
+            }
 
-        if (this.dataSetToChange !== undefined) {
-            let newRowList = []
-            const rowList = this.specificSettingsRows
-            newRowList = rowList.filter(row => row.id !== newDataSetRow.id)
-            newRowList.push(newDataSetRow)
-            this.specificSettingsRows = newRowList
+            if (this.dataSetToChange !== undefined) {
+                let newRowList = []
```

Comment:
This is a minor change, but this appears to be more complicated than it needs to be.  I think it could look like this:

```js
this.specificSettingsRows = this.specificSettingsRows.filter(row => row.id !== newDataSetRow.id)
this.specificSettingsRows.push(newDataSetRow)
```

Reply:
I applied this change also to programs specificSettings